### ISSUE TITLE
Variable referenced in multiple fields causes issues in Fusion

### ISSUE THREAD

**Issue by sergiu-enache** [ChilliCream/graphql-platform#6554]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Product

Hot Chocolate

### Describe the bug

If same variable is referenced in multiple fields of subgraph is forwarded multiple times by Gateway (from the looks of query plan)

` "forwardedVariables": [
    {
      "variable": "test"
    },
    {
      "variable": "test"
    }
  ]`

and results in error: 
`"errors": [
    {
      "message": "An item with the same key has already been added. Key: test",
      "extensions": {
        "code": "HC0012"
      }
    }
  ]`

### Steps to reproduce

1.  Reference same variable in multiple fields of subgraph

### Version

13.5.1

**Comment by interad-woergoetter** [comment]
I'm facing the same issue

**Comment by interad-woergoetter** [comment]
I think the problem is, that one of the arguments is nullable, and the other one is not.

**Comment by sergiu-enache** [comment]
you are correct @interad-woergoetter I guess a workaround is to duplicate variable as nullable until HC-13.6.0 stable is published
